app/gui2/src/components/widgets/NumericInputWidget.vue

```diff
@@ -14,16 +14,28 @@ const emit = defineEmits<{
   input: [content: string]
 }>()
 
-const inputFieldActive = ref(false)
-// Edited value reflects the `modelValue`, but does not update it until the user defocuses the field.
-const editedValue = ref(`${props.modelValue}`)
+const DEFAULT_PLACEHOLDER = ''
+const SLIDER_INPUT_THRESHOLD = 4.0
+const MIN_CONTENT_WIDTH = 56
+
+const initialValue = ref('')
+const placeholder = ref('')
 watch(
   () => props.modelValue,
   (newValue) => {
-    editedValue.value = `${newValue}`
+    const newNumber = typeof newValue === 'number' ? newValue : parseFloat(newValue)
+    if (!Number.isNaN(newNumber)) {
+      initialValue.value = `${newValue}`
+    } else {
+      placeholder.value = `${newValue}`
+    }
```

Comment:
When `v-model` changes from non-numeric to numeric, we still remember the old placeholder - I don't think this is desired. We could use just `computed`, couldn't we?

Also, I think this logic should belong to `WidgetNumber`. Here, as we define the "general purpose" numeric input, we should allow specifying any placeholder (including numeric one).